test/qpext/test_qpext.py

```diff
@@ -0,0 +1,98 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import logging
+import time
+import requests
+from urllib.parse import urlparse
+from kubernetes import client
+from kserve import (
+    constants,
+    KServeClient,
+    V1beta1InferenceService,
+    V1beta1InferenceServiceSpec,
+    V1beta1PredictorSpec,
+    V1beta1SKLearnSpec,
+)
+from kubernetes.client import V1ResourceRequirements
+import pytest
+
+from ..common.utils import predict, get_cluster_ip
+from ..common.utils import KSERVE_TEST_NAMESPACE
+
+logging.basicConfig(level=logging.INFO)
+
+ENABLE_METRIC_AGG = "serving.kserve.io/enable-metric-aggregation"
+METRICS_AGG_PORT = "9088"
+METRICS_PATH = "/metrics"
+
+
+@pytest.mark.fast
+def test_qpext_kserve():
+    # test the qpext using the sklearn predictor
+
+    service_name = "isvc-sklearn-v2"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        sklearn=V1beta1SKLearnSpec(
+            storage_uri="gs://seldon-models/sklearn/mms/lr_model",
+            protocol_version="v2",
+            resources=V1ResourceRequirements(
+                requests={"cpu": "50m", "memory": "128Mi"},
+                limits={"cpu": "100m", "memory": "512Mi"},
+            ),
+        ),
+    )
+
+    isvc = V1beta1InferenceService(
+        api_version=constants.KSERVE_V1BETA1,
+        kind=constants.KSERVE_KIND,
+        metadata=client.V1ObjectMeta(
+            name=service_name, namespace=KSERVE_TEST_NAMESPACE,
+            # set the metric aggregation annotation to true
+            annotations={ENABLE_METRIC_AGG: True},
+        ),
+        spec=V1beta1InferenceServiceSpec(predictor=predictor),
+    )
+
+    kserve_client = KServeClient(config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    res = predict(service_name, "./data/iris_input_v2.json", protocol_version="v2")
+    assert res["outputs"][0]["data"] == [1, 1]
+
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+def send_metrics_request(ks_client, service_name):
+    isvc = ks_client.get(
+        service_name,
+        namespace=KSERVE_TEST_NAMESPACE,
+        version="v2",
+    )
+    # temporary sleep until this is fixed https://github.com/kserve/kserve/issues/604
+    time.sleep(10)
+    cluster_ip = get_cluster_ip()
+    host = urlparse(isvc["status"]["url"]).netloc
+    headers = {"Host": host}
+
+    url = f"http://{cluster_ip}:{METRICS_AGG_PORT}/{METRICS_PATH}"
```

Comment:
I think this will not work as `cluster_ip` is for the ingress gateway which does not listen on metrics agg port.

Reply:
yeah you are right, I just copy pasted what the other tests did without realizing. I can get the IP from the pod. Not sure if local host_ip will work though. I dont think we can port-forward with how the test is set up